crates/matrix-sdk-base/src/client.rs

```diff
@@ -86,71 +86,30 @@ pub type Token = String;
 /// accordingly updates its state.
 #[derive(Clone)]
 pub struct BaseClient {
-    /// The current client session containing our user id, device id and access
-    /// token.
-    session: Arc<RwLock<Option<Session>>>,
     /// The current sync token that should be used for the next sync call.
     pub(crate) sync_token: Arc<RwLock<Option<Token>>>,
     /// Database
     store: Store,
+    /// The store used for encryption
+    #[cfg(feature = "e2e-encryption")]
+    crypto_store: Arc<dyn CryptoStore>,
```

Comment:
Why were all places where a `CryptoStore` is handled converted to use `Arc` instead of `Box`? We require them to be `Sync` and `Send` which usually means that they are already behind an `Arc` or something else that's easily cloneable.

Reply:
See my other comment. I couldn't figure out how to link a line in the source code.